docs/content/config/security/rspamd.md

```diff
@@ -100,6 +100,15 @@ DMS does not supply custom values for DNS servers to Rspamd. If you need to use
 
     If you want to see an environment variable (like `RSPAMD_DNS_SERVERS`) to support custom DNS servers for Rspamd being added to DMS, please raise a feature request issue.
 
+!!! warning
+
+    Rspamd heavily relies on a functioning DNS. In case your DNS does not work, you will encounter issues in the form of
+
+    1. e-mails being marked as spam when they actually are not spam;
+    2. e-mails being rejected because Rspamd is confident that the e-mails are spam, when they actually are not spam.
+
+    An example here are the SPF, DKIM and DMARC checks, all of which are based on DNS records. These checks have a symbol for DNS temporary errors, with a non-zero weight - i.e., they will increase the spam score of the e-mail. This is undesirable, especially because mechanisms like SPF, DKIM and DMARC are frequently used to avoid spam.
```

Comment:
```suggestion
!!! warning

    Rspamd heavily relies on DNS. If your DNS is misconfigured, you will encounter issues when Rspamd queries DNS to assess if mail is spam. Legitimate mail is then unintentionally marked as spam or worse, rejected entirely.

    When Rspamd is deciding if mail is spam, it will check DNS records for SPF, DKIM and DMARC. Each of those has an associated symbol for DNS temporary errors with a non-zero weight assigned. That weight contributes towards the spam score assessed by Rspamd which is normally desirable - provided your network DNS is functioning correctly, otherwise when DNS is broken all mail is biased towards spam due to these failed DNS lookups.
```

Reworded that a bit, how's that sound? The two ordinal points seemed roughly the same so I collapsed them.

Looking at the earlier context it seems to be more about routing of DNS queries? Such as setting `1.1.1.1` / `8.8.8.8` or your own local resolver?

My rewrite above probably need another revision in the first paragraph at least to clarify that distinction as it implies the DNS records themselves not being configured correctly, rather than the lookups failing because the DNS could not resolve/reach the actual nameserver? (_I tried to clarify that a bit better at the end_)

Reply:
I intended to make users aware that a properly functioning DNS server and the correct corresponding configuration to this DNS server are important.

It's really not so much about correctness properties of DNS records themselves.

```suggestion
!!! warning

    Rspamd heavily relies on a properly working DNS server that it can use to resolve DNS queries. If your DNS server is misconfigured, you will encounter issues when Rspamd queries DNS to assess if mail is spam. Legitimate mail is then unintentionally marked as spam or worse, rejected entirely.

    When Rspamd is deciding if mail is spam, it will check DNS records for SPF, DKIM and DMARC. Each of those has an associated symbol for DNS temporary errors with a non-zero weight assigned. That weight contributes towards the spam score assessed by Rspamd which is normally desirable - provided your network DNS is functioning correctly, otherwise when DNS is broken all mail is biased towards spam due to these failed DNS lookups.
```